lib/sdm/maven.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GitHubRepoRef } from "@atomist/automation-client";
+import { scanFreePort } from "@atomist/automation-client/lib/util/port";
+import {
+    DoNotSetAnyGoals,
+    execPromise,
+    GeneratorRegistration,
+    goal,
+    hasFile,
+    not,
+    SdmGoalState,
+    slackSuccessMessage,
+} from "@atomist/sdm";
+import {
+    configure,
+    Version,
+} from "@atomist/sdm-core";
+import { Build } from "@atomist/sdm-pack-build";
+import {
+    mavenBuilder,
+    MavenProjectVersioner,
+    MvnVersion,
+    SpringProjectCreationParameterDefinitions,
+    SpringProjectCreationParameters,
+    TransformMavenSpringBootSeedToCustomProject,
+} from "@atomist/sdm-pack-spring";
+import {
+    codeLine,
+    url,
+} from "@atomist/slack-messages";
+import { UpdateReadmeTitle } from "../transform/updateReadmeTitle";
+
+/**
+ * Atomist SDM Sample
+ * @description SDM to create and build Maven projects
+ * @tag sdm,generator,maven
+ * @instructions <p>Now that the SDM is up and running, create a new Maven
```

Comment:
Instructions talk about docker and the need for docker to be available.

Reply:
Removed.